src/frontend/web_application/src/components/SignupForm/presenter.jsx

```diff
@@ -59,6 +66,19 @@ class SignupForm extends Component {
     this.setState(generateStateFromProps(newProps));
   }
 
+  handleOpenModal() {
+    this.setState(prevState => ({
```

Comment:
if you don't need prevState for specific condition, you can omit it:

```
this.setState({ isModalOpen: true });
```

This will not affect the rest of the state (yes I know it's a bit weird).